backend/app/processing/db_storage.py

```
@@ -0,0 +1,57 @@
+        for f in os.listdir(db_dir):
+            if "_" in f:
+                parts = f.split("_", 1)
+                if len(parts) == 2:
+                    existing[parts[1]] = f
+        for orig, fname in existing.items():
+            if orig not in keep_filenames:
+                (db_dir / fname).unlink()
+                removed.append(orig)
+        return removed
+
+    def save_uploaded_file(self, chatbot_id: str, file_obj, original_filename: str) -> Path:
```

Ruff (0.8.2)

Missing type annotation for function argument `file_obj`

(ANN001)